File: ios/ApiVideoUploader.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ApiVideoUploader, NSObject)

RCT_EXTERN_METHOD(upload:(NSString)token :(NSString)filename :(NSString)filepath
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/ApiVideoUploader.swift
import VideoUploaderIos

@objc(ApiVideoUploader)
class ApiVideoUploader: NSObject {
    
    @objc(upload:::withResolver:withRejecter:)
    func upload(token: String, filename: String, filepath: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        
        let uploader = VideoUploader()
        let parsed = filepath.replacingOccurrences(of: "file://", with: "")
        let url = URL(fileURLWithPath: parsed)
        var jsonString = ""
        var jsonVideo: Dictionary<String, AnyObject>?
        uploader.uploadWithDelegatedToken(delegatedToken: token, fileName: filename, filePath: filepath, url: url){ (json, error) in
            if(error == nil){
                jsonVideo = json
                if let jsonData = try? JSONSerialization.data(
                    withJSONObject: json as Any,
                    options: []) {
                    jsonString = String(data: jsonData,
                                        encoding: .ascii)!
                }
            }else{
                jsonString = error.debugDescription
            }
        }
        resolve(jsonString)
    }

}
